main.go

```diff
@@ -2,39 +2,115 @@ package main
 
 import (
 	"fmt"
+	//"math/rand/v2"
+	"math/rand"
+	"sync"
+	"time"
 )
 
 const (
 	SIZE   = 100_000_000
 	CHUNKS = 8
 )
 
+var wg sync.WaitGroup
+
 // generateRandomElements generates random elements.
 func generateRandomElements(size int) []int {
-	// ваш код здесь
+	var slice []int
+
+	if size <= 0 {
+		return []int{} // empty slice
+	}
+
+	randSource := rand.NewSource(time.Now().UnixNano())
+	randRange := rand.New(randSource)
+
+	for range size {
+		//randomNumber := (rand.IntN(900_000_000) + 1)  //через /rand/2 было бы проще, но в задании хотят именно /rand
+		randomNumber := randRange.Intn(900_000_000)
+		slice = append(slice, randomNumber)
+	}
+
+	return slice
 }
 
 // maximum returns the maximum number of elements.
 func maximum(data []int) int {
-	// ваш код здесь
+
+	max := 0
+
+	if len(data) == 0 {
+		return max
+	}
+
+	if len(data) == 1 {
+		return data[0]
+	}
+
+	for _, v := range data {
+		if v > max {
+			max = v
+		}
+	}
+	return max
 }
 
 // maxChunks returns the maximum number of elements in a chunks.
 func maxChunks(data []int) int {
-	// ваш код здесь
+	partOfSlice := len(data) / CHUNKS
+	max := 0
+
+	if len(data) == 0 {
+		return max
+	}
+
+	if len(data) == 1 {
+		return data[0]
+	}
+
+	resultChan := make(chan int, CHUNKS) // буферизированный канал для сбора результатов
+	wg.Add(CHUNKS)
+	for i := 0; i < CHUNKS; i++ {
+		firstIndex := i * partOfSlice
+		lastIndex := firstIndex + partOfSlice
+		preparedSlice := data[firstIndex:lastIndex]
+
+		go func(slice []int) {
+			defer wg.Done()
+			maxInPreparedSlice := maximum(slice)
+			resultChan <- maxInPreparedSlice // отправляем результат в канал
```

Comment:
Можно сразу в массив писать
```suggestion
			results[i] = maxVal
```